pkg/cmd/deadman/operator_condition_challenger.go

```diff
@@ -0,0 +1,179 @@
+		clusterOperatorLister: configInformers.Config().V1().ClusterOperators().Lister(),
+
+		queue: workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "OperatorConditionChallenger"),
+	}
+
+	configInformers.Config().V1().ClusterOperators().Informer().AddEventHandlerWithResyncPeriod(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				clusterOperator := obj.(*configv1.ClusterOperator)
+				c.queue.Add(clusterOperator.Name)
+			},
+			UpdateFunc: func(oldObj, newObj interface{}) {
+				clusterOperator := newObj.(*configv1.ClusterOperator)
+				c.queue.Add(clusterOperator.Name)
+			},
+		},
+		1*time.Minute,
+	)
+
+	return c
+}
+
+func getMostRecentConditionTransitionTime(clusterOperator *configv1.ClusterOperator) *time.Time {
+	if len(clusterOperator.Status.Conditions) == 0 {
+		return nil
+	}
+
+	newestTime := time.Time{}
+	for _, condition := range clusterOperator.Status.Conditions {
+		if condition.LastTransitionTime.After(newestTime) {
+			newestTime = condition.LastTransitionTime.Time
+		}
+	}
+	return &newestTime
+}
+
+func (c *OperatorConditionChallenger) syncHandler(ctx context.Context, key string) error {
+	clusterOperator, err := c.clusterOperatorLister.Get(key)
+	if apierrors.IsNotFound(err) {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	if isCheckingForStaleness(clusterOperator) {
+		// staleness check is in progress already, just return
+		return nil
+	}
+
+	mostRecentOperatorUpdate := getMostRecentConditionTransitionTime(clusterOperator)
+	if mostRecentOperatorUpdate == nil {
+		// this means we have no conditions, so there's no work to do.
+		return nil
+	}
+	nextCheckTimeBasedOnOperatorUpdate := mostRecentOperatorUpdate.Add(c.durationAllowedBetweenStalenessChecks)
+
+	// check to see if the next time to check for activity needs to be updated based on the data.
+	cachedNextCheckTime, ok := c.operatorToTimeToCheckForStaleness[clusterOperator.Name]
+	switch {
+	case !ok:
+		c.operatorToTimeToCheckForStaleness[clusterOperator.Name] = nextCheckTimeBasedOnOperatorUpdate
+
+	case cachedNextCheckTime.Before(nextCheckTimeBasedOnOperatorUpdate):
+		c.operatorToTimeToCheckForStaleness[clusterOperator.Name] = nextCheckTimeBasedOnOperatorUpdate
+
+	}
+
+	now := time.Now()
+	nextLivenessCheckTime := c.operatorToTimeToCheckForStaleness[clusterOperator.Name]
+	if now.Before(nextLivenessCheckTime) {
+		return nil
+	}
+
+	// if we're past the deadline since our last check, so we will perform a staleness check by resetting the message.
+	// only modify the message to avoid changing machine-readable reasons or status in the common case when operators
+	// are properly managing their status.
+	clusterOperatorToWriteAsStale := clusterOperator.DeepCopy()
+	for i, condition := range clusterOperator.Status.Conditions {
+		clusterOperatorToWriteAsStale.Status.Conditions[i].Message = "Checking for stale status, the active operator will reset this message: " + condition.Message
```

Comment:
I think if we don't want to add a special field into the Condition type that indicates the staleness, to be more explicit.

Reply:
> I think if we don't want to add a special field into the Condition type that indicates the staleness, to be more explicit.

Conditions are standardized in metav1.  Our type pre-dates metav1, but we should remain consistent and not add any fields.

Reply:
👍